### PR TITLE
Add a transformation to convert all the columns of a DataFrame to Par…

### PR DESCRIPTION
**Pull request by MrPowers**
…quet compatible column names

### CODE REVIEW

**Comment by MrPowers** [comment]
As suggested in this issue: https://github.com/MrPowers/spark-daria/issues/115

@nvander1 @snithish @gorros - can you please take a look and let me know if this looks good.  Any other edge cases we can cover in the test?  Thanks!

File: src/test/scala/com/github/mrpowers/spark/daria/sql/TransformationsTest.scala [hunk 1]
```diff
@@ -757,9 +755,36 @@ object TransformationsTest extends TestSuite with DataFrameComparer with ColumnC
         // HACK - don't have a good way to test this yet
         // Need to add assertStructTypeColumnEquality: https://github.com/MrPowers/spark-fast-tests/issues/38
         // Or update assertSmallDataFrameEquality to work with DataFrames that have StructType columns: https://github.com/MrPowers/spark-fast-tests/issues/37
-
       }
+    }
```
Comment (gorros) [inline src/test/scala/com/github/mrpowers/spark/daria/sql/TransformationsTest.scala]: @eclosson maybe it also makes sense to write dataframe with invalid column name as parquet and make sure that if fails,and then transform and make sure it works.

**Comment by gorros** [comment]
I've reviewed it. Added a comment regarding testing.

**Comment by snithish** [comment]
@MrPowers I was thinking what if the "find" and "replace" strings become params and then they be hardcoded as constants.. meaning having a generic column name replace taking param on how to do it.. It will certainly make it flexible for any format and make it easy to extend to. 
What are your thought?

**Comment by MrPowers** [comment]
@snithish - yep, like that feedback, will refactor and send you a new version.  Love how you think so abstractly ;)

@gorros - I will add that test, great point.

@eclosson - Right now we're thinking about the characters that we want to remove from a string... I think we might want to actually think about the characters we want to keep instead.  All we want are letters, numbers, and underscores right?  I am thinking about how we can make this code even more robust to handle weird characters that aren't explicitly blacklisted.  For example, our code isn't currently factoring in `\r`.  I will dig into this.

**Comment by MrPowers** [comment]
@gorros - added the test you suggested BTW ;)
